cli/client.go

```diff
@@ -1355,6 +1357,90 @@ func (c *viamClient) machinesPartCopyFilesAction(
 	return nil
 }
 
+type robotsPartTunnelArgs struct {
+	Organization    string
+	Location        string
+	Machine         string
+	Part            string
+	LocalPort       int
+	DestinationPort int
+}
+
+// RobotsPartTunnelAction is the corresponding Action for 'machines part tunnel'.
+func RobotsPartTunnelAction(c *cli.Context, args robotsPartTunnelArgs) error {
+	client, err := newViamClient(c)
+	if err != nil {
+		return err
+	}
+
+	return client.robotPartTunnel(c, args)
+}
+
+func tunnelTraffic(ctx *cli.Context, robotClient *client.RobotClient, local, dest int) error {
+	li, err := net.Listen("tcp", net.JoinHostPort("localhost", strconv.Itoa(local)))
+	if err != nil {
+		return fmt.Errorf("failed to create listener %w", err)
+	}
+	infof(ctx.App.Writer, "tunneling connections from local port %v to destination port %v on machine part...", local, dest)
+	defer func() {
+		if err := li.Close(); err != nil {
+			warningf(ctx.App.ErrWriter, "error closing listener: %s", err)
+		}
+	}()
+
+	var wg sync.WaitGroup
+	for {
+		if ctx.Err() != nil {
+			break
+		}
+		conn, err := li.Accept()
+		if err != nil {
+			warningf(ctx.App.ErrWriter, "failed to accept connection: %s", err)
+			continue
+		}
+
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			// call tunnel once per connection, the connection passed in will be closed
+			// by Tunnel.
+			if err := robotClient.Tunnel(ctx.Context, conn, dest); err != nil {
+				printf(ctx.App.Writer, "error while tunneling connection: %s", err)
+			}
+		}()
+	}
+	wg.Wait()
+	return nil
+}
+
+func (c *viamClient) robotPartTunnel(cCtx *cli.Context, args robotsPartTunnelArgs) error {
+	orgStr := args.Organization
+	locStr := args.Location
+	robotStr := args.Machine
+	partStr := args.Part
+
+	// Create logger based on presence of debugFlag.
+	logger := logging.FromZapCompatible(zap.NewNop().Sugar())
+	globalArgs, err := getGlobalArgs(cCtx)
+	if err != nil {
+		return err
+	}
+	if globalArgs.Debug {
+		logger = logging.NewDebugLogger("cli")
+	}
+
+	dialCtx, fqdn, rpcOpts, err := c.prepareDial(orgStr, locStr, robotStr, partStr, globalArgs.Debug)
```

Comment:
I would expect we'd only need a `partStr`. I'm not sure where along the refactor we are with simplifying the CLI. @stuqdog ?

Reply:
That refactor is in already! We continue to let people pass org/loc/machine along with part because 1) backwards compatibility, and 2) technically more than part is needed if someone wants to pass by name rather than ID. 

I'm still fully reviewing but on first glance I think this looks correct to me. Only `part` is listed as a required argument in the CLI command so a user need only pass a part ID to successfully tunnel, but could pass `my-org`, `my-location`, `my-robot`, and `my-part` for a fully qualified, ID-less experience as well.

Reply:
Got it! So this `prepareDial` is what's validating that we have enough inputs filled out?

> technically more than part is needed if someone wants to pass by name rather than ID

Ah, I never realized we supported using the given "names". Do `robot`s have names as well? Maybe it's just the same string as the main part's name?

Do locations and orgs also have both a short-name string form and a UUID form?

Reply:
> Got it! So this `prepareDial` is what's validating that we have enough inputs filled out?
More or less. I think technically the validation happens in the `robotPart` call, but `prepareDial` needs to get a part to get the `fqdn`.

> Ah, I never realized we supported using the given "names". Do robots have names as well? Maybe it's just the same string as the main part's name?

Yep! So e.g., I can run `viam robot status --organization "Ethan's Org" --location "ethan office" --machine "webrtc-test-main"` and get correct results